lib/my_john_deere_api/net_http_retry/invalid_response_error.rb

```diff
@@ -0,0 +1,23 @@
+module MyJohnDeereApi
+  module NetHttpRetry
+    ##
+    # This error is used when a single request has exceeded
+    # the number of retries allowed by NetHttpRetry::Decorator::MAX_RETRIES.
+
+    class InvalidResponseError < StandardError
+
+      ##
+      # argument is a string which describes the attempted request
+
+      def initialize(response)
+        message = {
+          code: response.code,
+          message: response.message,
+          body: response.body,
+        }.to_json
```

Comment:
If you just want to return a subset of `response`, you could use `slice` here.
```suggestion
        message = response.slice(:code, :message, :body).to_json
```

Reply:
response is not a hash, would that still work?

Reply:
I've used it with ActiveRecord objects, but I guess that might be some Rails magic rather than a standard Ruby behavior.

Reply:
Well shoot, that's a Rails thing: https://api.rubyonrails.org/classes/ActiveRecord/Core.html#method-i-slice